code/datums/mapgen/planetary/AsteroidGenerator.dm

```diff
@@ -0,0 +1,169 @@
+			BIOME_LOW_HUMIDITY = /datum/biome/asteroid/carp,
+			BIOME_MEDIUM_HUMIDITY = /datum/biome/asteroid/carp,
+			BIOME_HIGH_HUMIDITY = /datum/biome/asteroid/carp,
+			BIOME_HIGHEST_HUMIDITY = /datum/biome/asteroid/carp
+		),
+		BIOME_HOTTEST = list(
+			BIOME_LOWEST_HUMIDITY = /datum/biome/asteroid/carp,
+			BIOME_LOW_HUMIDITY = /datum/biome/asteroid,
+			BIOME_MEDIUM_HUMIDITY = /datum/biome/asteroid,
+			BIOME_HIGH_HUMIDITY = /datum/biome/asteroid/carp,
+			BIOME_HIGHEST_HUMIDITY = /datum/biome/asteroid/carp //gee what a diverse place
+		)
+	)
+
+	cave_biome_table = list(
+		BIOME_COLDEST_CAVE = list(
+			BIOME_LOWEST_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_LOW_HUMIDITY = /datum/biome/cave/asteroid/ice,
+			BIOME_MEDIUM_HUMIDITY = /datum/biome/cave/asteroid/ice,
+			BIOME_HIGH_HUMIDITY = /datum/biome/cave/asteroid/ice,
+			BIOME_HIGHEST_HUMIDITY = /datum/biome/cave/asteroid/ice
+		),
+		BIOME_COLD_CAVE = list(
+			BIOME_LOWEST_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_LOW_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_MEDIUM_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_HIGH_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_HIGHEST_HUMIDITY = /datum/biome/cave/asteroid/ice
+		),
+		BIOME_WARM_CAVE = list(
+			BIOME_LOWEST_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_LOW_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_MEDIUM_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_HIGH_HUMIDITY = /datum/biome/cave/asteroid/carp_den,
+			BIOME_HIGHEST_HUMIDITY = /datum/biome/cave/asteroid/carp_den
+		),
+		BIOME_HOT_CAVE = list(
+			BIOME_LOWEST_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_LOW_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_MEDIUM_HUMIDITY = /datum/biome/cave/asteroid/carp_den,
+			BIOME_HIGH_HUMIDITY = /datum/biome/cave/asteroid/carp_den,
+			BIOME_HIGHEST_HUMIDITY = /datum/biome/cave/asteroid/carp_den
+		)
+	)
+
+/datum/biome/asteroid
+	open_turf_types = list(
+	/turf/open/space = 10
+	)
+
+/datum/biome/asteroid/carp
+	mob_spawn_list = list(
+	/mob/living/simple_animal/hostile/carp = 1
+		)
+
+/datum/biome/cave/asteroid
+	closed_turf_types =  list(
+	/turf/closed/mineral/random = 1
+	)
+	open_turf_types = list(
+	/turf/open/floor/plating/asteroid/airless = 1
+	)
+
+/datum/biome/cave/asteroid/vanilla
+	flora_spawn_list = list(
+	/obj/structure/flora/ash/space/voidmelon = 1,
+	/obj/structure/flora/rock = 1,
+	/obj/structure/flora/rock/pile = 1
+	)
+
+	feature_spawn_list = list(
+	/obj/structure/geyser/random = 1
+	)
+
+	mob_spawn_list = list(
+	/mob/living/simple_animal/hostile/asteroid/goliath = 25, /obj/structure/spawner/mining/goliath = 30, \
+		/mob/living/simple_animal/hostile/asteroid/basilisk = 25, /obj/structure/spawner/mining = 30, \
+		/mob/living/simple_animal/hostile/asteroid/hivelord = 25, /obj/structure/spawner/mining/hivelord = 30, \
+		/mob/living/simple_animal/hostile/asteroid/goldgrub = 10
```

Comment:
```suggestion
		 /obj/structure/geyser/random = 1,
		 /obj/structure/spawner/mining/goliath = 1,
		 /obj/structure/spawner/mining = 1,
		 /obj/structure/spawner/mining/hivelord = 1
	)

	mob_spawn_list = list(
		/mob/living/simple_animal/hostile/asteroid/goliath = 25, 
		/mob/living/simple_animal/hostile/asteroid/basilisk = 25, 
		/mob/living/simple_animal/hostile/asteroid/hivelord = 25, 
		/mob/living/simple_animal/hostile/asteroid/goldgrub = 10
```

See lower comment on making the spawner presets for regions. Have the spawners be features so they don't stack up oppressively